### DISCUSSION TITLE
Dynamic route does not support .  (dot)

### DISCUSSION THREAD

**Discussion by lucky-bit** [livebud/bud#112] [Ideas]
1. Define get route:
  router.Get(`/:id`, controller.Show)
2. Define id as string
 // Show page
 // GET /:id
 func (c *Controller) Show(ctx context.Context, id string) (story *Story, err error) {
3. Navigate to http://127.0.0.1:3000/some.value
===> 404 page not found

**Comment by matthewmueller** [comment]
Thanks for trying out Bud @lucky-bit! 

I'd like to add support for this, but it will be more about specifying a file format, e.g. `http://127.0.0.1:3000/some.json` or `http://127.0.0.1:3000/some.rss` triggering the `Some` action. Does that work for your use case?

One other question:

> Define get route:

I'm assuming you defined this in the `controller/` package or did you use the router package manually?

**Comment by lucky-bit** [comment]
Hey @matthewmueller!
Unfortunately file format and triggering action won't cover my case. 
I would like to present order details according to provided order id such as xyz.01234567 by visiting http://127.0.0.1:3000/xyz.01234567

> I'm assuming you defined this in the `controller/` package or did you use the router package manually?

No manual work, route created using _bud new controller_

**Comment by matthewmueller** [comment]
Ah, that's an interesting one. So basically `/orders/:id`, where `:id` is `xyz.01234567`? Out of curiousity, why are dots a requirement for an order id? Existing database?

At first glance, I'd probably follow what Rails decided to do in this case. Maybe that means defining a custom route?

_(Converting this to a discussion since it's currently working as expected, though we might want to change this)_

**Reply by lucky-bit** [reply]
> Existing database?

Yes

> Maybe that means defining a custom route?

Sounds good, any time lines for this?

**Reply by matthewmueller** [reply]
Thanks @lucky-bit, I need to think about this one a bit more. 

> So basically /orders/:id, where :id is xyz.01234567? 

It seems like Rails does support this with custom regexps: https://prathamesh.tech/2020/06/15/allowing-dots-in-rails-routes/

That could be the way for Bud too.

> Sounds good, any time lines for this?

This is still a side project, so I can't commit to timelines. What I will say is that I'm working on wrapping up [v0.2](https://github.com/livebud/bud/discussions/18) hopefully by the end of the week, then I start on [v0.3](https://github.com/livebud/bud/discussions/21). 

This v0.3 will take ~1-3 months. After v0.3, there will be a stream of configuration flowing through Bud that we can use to attach custom routing on. I might finish custom routing as part of v0.3 or as a follow-up release.
